src/organizations/components/OrgProfileTab/DeletePaidOrgOverlay.tsx

```diff
@@ -0,0 +1,191 @@
+// Libraries
+import React, {FC, useContext, useState} from 'react'
+import {useDispatch, useSelector} from 'react-redux'
+import {
+  Alert,
+  AlignItems,
+  Button,
+  ComponentColor,
+  ComponentSize,
+  ComponentStatus,
+  FlexBox,
+  FlexDirection,
+  IconFont,
+  Input,
+  InputLabel,
+  InputType,
+  Overlay,
+} from '@influxdata/clockface'
+
+// Selectors
+import {
+  selectCurrentAccount,
+  selectCurrentOrg,
+  selectUser,
+} from 'src/identity/selectors'
+
+// API
+import {deleteOrganization} from 'src/identity/apis/org'
+
+// Constants
+import {CLOUD_URL} from 'src/shared/constants'
+
+// Notifications
+import {deleteOrgFailed} from 'src/shared/copy/notifications'
+import {notify} from 'src/shared/actions/notifications'
+
+// Utils
+import {OverlayContext} from 'src/overlays/components/OverlayController'
+import {reportErrorThroughHoneyBadger} from 'src/shared/utils/errors'
+import {setToLocalStorage} from 'src/localStorage'
+
+// Styles
+import './DeletePaidOrgOverlay.scss'
+
+// Required to avoid unwarranted console errors from clockface <Input /> when type=radio.
+const nullOnChange = () => null
+
+export const DeletePaidOrgOverlay: FC = () => {
+  const account = useSelector(selectCurrentAccount)
+  const org = useSelector(selectCurrentOrg)
+  const user = useSelector(selectUser)
+  const dispatch = useDispatch()
+
+  const {onClose} = useContext(OverlayContext)
+
+  const [deleteButtonStatus, setDeleteButtonStatus] = useState(
+    ComponentStatus.Disabled
+  )
+  const [userAcceptedTerms, setUserAcceptedTerms] = useState(false)
+
+  const toggleAcceptedTerms = () => {
+    const currentAcceptanceStatus = !userAcceptedTerms
+    setUserAcceptedTerms(currentAcceptanceStatus)
+
+    setDeleteButtonStatus(
+      currentAcceptanceStatus
+        ? ComponentStatus.Default
+        : ComponentStatus.Disabled
+    )
+  }
+
+  const handleDeleteOrg = async () => {
+    try {
+      setDeleteButtonStatus(ComponentStatus.Loading)
+
+      await deleteOrganization(org.id)
```

Comment:
does this need to be `await`ed? can some things be done immediately afterward? It seems like `setDeleteButtonStatus()` and `onClose()` can happen immediately. From a usability standpoint, it might be better to make the button disabled as soon as the request is made.

It might be worth changing this into a `.then().catch()` pattern, like:

```ts
deleteOrganization(org.id).then(() => {
  // Adding (and on reload, removing) this property from local storage
  // is a work-around so that it's possible for the user to be notified of org deletion
  // AFTER app reload. See src/components/notifications/Notifications.tsx,
  // which detects the property, and if found, sends a notification then deletes it.
  setToLocalStorage('justDeletedOrg', org.name)
  window.location.href = `${CLOUD_URL}`
}).catch(err => {
  dispatch(notify(deleteOrgFailed(org.name)))
  reportErrorThroughHoneyBadger(err, {
    name: 'Org deletion failed',
    context: {
      user,
      account,
      org,
    },
  })
  setDeleteButtonStatus(ComponentStatus.Disabled)
  setUserAcceptedTerms(false)
})

setDeleteButtonStatus(ComponentStatus.Default)
setDeleteButtonStatus()
```

that will let `setDeleteButtonStatus` and `setDeleteButtonStatus` get called immediately, then the `setToLocalStorage` and `window.location.href` happen after it succeeds

Reply:
Users cannot take any actions on a deleted org so we need to redirect the user to an existing org as soon as the delete request returns a successful response. My vote is to keep the modal open until the request completes. In addition to showing the button as loading, we should also prevent the modal from being dismissed until the request completes (and re-enable dismissal if there's an error returned).

Reply:
👍

Reply:
Thanks for the direction Gunnar